### PR TITLE
chore: change linter to Ruff

### LINKED ISSUE #211: Change linters to `ruff`
https://github.com/charliermarsh/ruff

### PR DESCRIPTION
**Pull request by Calychas**
Closes #211

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -112,19 +109,25 @@ line-length = 100
...
+select = [
+    "E",  # pycodestyle
+    "F",  # pyflakes
+    "UP", # pyupgrade
+    "D",  # pydocstyle
+    "I",  # isort
+    "B",   # flake8-bugbear
+    "NPY"  # NumPy
+]
+ignore = ["D212"]
```
Comment (RaczeQ) [inline pyproject.toml:124]: Old version had hits config:
```
add-select = ["D213"]
add-ignore = ["D202", "D212"]
```

File: pyproject.toml [hunk 2]
```diff
@@ -112,19 +109,25 @@ line-length = 100
 target-version = ["py38", "py39", "py310", "py311"]
 preview = true
 
-[tool.isort]
-profile = "black"
+[tool.ruff]
```
Comment (RaczeQ) [inline pyproject.toml:112]: Is there any flake8 config copy within pyproject.toml to configure ruff?

File: pyproject.toml (hunk 1 above)
Comment (Calychas) [reply pyproject.toml:124]: "D213" - is automatically selected by "D".
"D202" - I fixed errors with it and removed the ignore

File: pyproject.toml (hunk 2 above)
Comment (Calychas) [reply pyproject.toml:112]: I moved all the relevant flake8 configuration to pyproject.toml:
*` max-line-length` is just `line-length`
* moved `max-doc-length = 100` to pycodestyle
* fixed B028 errors
* E203 is missing from [the ruleset ](https://beta.ruff.rs/docs/rules)
* added more folder ignores

**Review by RaczeQ — APPROVED** [review]

**Pull request merged**